cert/cert.go

```diff
@@ -443,6 +479,34 @@ func (spec *Spec) Certificate() *x509.Certificate {
 	return spec.tr.Provider.Certificate()
 }
 
+// CertificateAge returns age of certificate associated with spec
+func (spec *Spec) CertificateAge() time.Duration {
+	c := spec.Certificate()
+	if c == nil {
+		return -1
+	}
+	now := time.Now()
+	return now.Sub(c.NotBefore)
+}
+
+// CA age returns age of CA associated with Spec
+func (spec *Spec) CAAge() time.Duration {
+	c := spec.CA.GetPEM()
+	if c == nil {
+		return -1
+	}
+	certPem, _ := pem.Decode(c)
+	if certPem == nil {
+		return -1
+	}
+	parsedCert, err := x509.ParseCertificate(certPem.Bytes)
+	if err != nil {
+		return -1
+	}
```

Comment:
nit: Some debug-level logging may be useful here for trying to determine why this returned what it did, in each of the failure cases above.

Reply:
Agreed. I'll add some debug logging for each failure case here. 

However, there are some valid cases where CertificateAge() and CAAge() will return -1... for example, the path to a CA file is not required in the spec, so if this isn't set, then CAAge() will return -1. 

Anyway, I'll add the log statements. If a -1 is returned, the "cert_watching" metric with the "ca_age" or "cert_age" label will be -1. We can decide how important this is to alert on.

Reply:
Agreed regarding capturing information about the parse failure- minimally I'd view it as a warning level log line.

I've not checked if certmgr validates that it's writing parsing x509, but a warning at this level would also give us some correlation if certmgr ever manages to have a bug like I mentioned.